src/audio/mux/mux.c

```diff
@@ -711,13 +905,22 @@ static int mux_trigger(struct comp_dev *dev, int cmd)
 	 */
 	src_n_active = mux_source_status_count(dev, COMP_STATE_ACTIVE);
 	src_n_paused = mux_source_status_count(dev, COMP_STATE_PAUSED);
-
-	switch (cmd) {
-	case COMP_TRIGGER_PRE_START:
-		if (src_n_active || src_n_paused)
-			return PPL_STATUS_PATH_STOP;
+	if (dir == SOF_IPC_STREAM_PLAYBACK) {
```

Comment:
Isn't this changing the functionality for IPC3 as well as this code is only entered if direction is playback?

Reply:
I don't know all the details but it is strange to this question "resolved" without any answer.

Please resolve/hide only comments where it is obvious for everyone that all parties have come to 100% agreement and understanding. Even then it's sometimes good to leave some interesting discussions and rationales visible. Of course hide stuff like "missing new line".

There is no policy in this project to resolve all comments before merging. It is often useful to read some code review discussions even after the code is merged and resolved comments get in the way of that.